MAUI/DataGrid/Editing.md

```diff
@@ -0,0 +1,376 @@
+            RaisePropertyChanged("LastName");
+        }
+    }
+
+    public string Gender
+    {
+        get { return _gender; }
+        set
+        {
+            this._gender = value;
+            RaisePropertyChanged("Gender");
+        }
+    }
+
+    public string ShipCity
+    {
+        get { return _shipCity; }
+        set
+        {
+            this._shipCity = value;
+            RaisePropertyChanged("ShipCity");
+        }
+    }
+
+    public string ShipCountry
+    {
+        get { return _shipCountry; }
+        set
+        {
+            this._shipCountry = value;
+            RaisePropertyChanged("ShipCountry");
+        }
+    }
+
+    public string Freight
+    {
+        get { return _freight; }
+        set
+        {
+            this._freight = value;
+            RaisePropertyChanged("Freight");
+        }
+    }
+
+    public DateTime ShippingDate
+    {
+        get { return _shippingDate; }
+        set
+        {
+            this._shippingDate = value;
+            RaisePropertyChanged("ShippingDate");
+        }
+    }
+
+    #endregion
+
+    #region INotifyPropertyChanged implementation
+
+    public event PropertyChangedEventHandler PropertyChanged;
+
+	private void RaisePropertyChanged (String Name)
+	{
+		if (PropertyChanged != null)
+			this.PropertyChanged (this, new PropertyChangedEventArgs (Name));
+	}
+
+    private Dictionary<string, object> storedValues;
+
+
+    public void BeginEdit()
+    {
+        this.storedValues = this.BackUp();
+    }
+
+    public void CancelEdit()
+    {
+        if (this.storedValues == null)
+            return;
+
+        foreach (var item in this.storedValues)
+        {
+            var itemProperties = this.GetType().GetTypeInfo().DeclaredProperties;
+            var pDesc = itemProperties.FirstOrDefault(p => p.Name == item.Key);
+            if (pDesc != null)
+                pDesc.SetValue(this, item.Value);
+        }
+    }
+
+    public void EndEdit()
+    {
+        if (this.storedValues != null)
+        {
+            this.storedValues.Clear();
+            this.storedValues = null;
+        }
+        Debug.WriteLine("End Edit Called");
+    }
+
+    protected Dictionary<string, object> BackUp()
+    {
+        var dictionary = new Dictionary<string, object>();
+        var itemProperties = this.GetType().GetTypeInfo().DeclaredProperties;
+        foreach (var pDescriptor in itemProperties)
+        {
+            if (pDescriptor.CanWrite)
+                dictionary.Add(pDescriptor.Name, pDescriptor.GetValue(this));
+        }
+        return dictionary;
+    }
+
+    #endregion
+}
+{% endhighlight %}
+
+
+## Editing events
+
+The SfDataGrid triggers the following events while editing:
+
+### CurrentCellBeginEdit
+
+The [SfDataGrid.CurrentCellBeginEdit]() event occurs when the CurrentCell enters into edit mode. The [DataGridCurrentCellBeginEditEventArgs]() has the following members which provides information for `SfDataGrid.CurrentCellBeginEdit` event:
```

Comment:
The [SfDataGrid.CurrentCellBeginEdit]() event occurs when the CurrentCell enters into edit mode. The [DataGridCurrentCellBeginEditEventArgs]() has the following members that provides information for `SfDataGrid.CurrentCellBeginEdit` event:

Reply:
![image](https://github.com/syncfusion-content/maui-docs/assets/113881396/d9c88639-8cfc-47e6-96ce-f5826648afab)